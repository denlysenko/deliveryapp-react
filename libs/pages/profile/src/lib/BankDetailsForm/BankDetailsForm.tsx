import React from 'react';

import { FormikHandlers } from 'formik';

import { InputText } from 'primereact/inputtext';
import { InputMask } from 'primereact/inputmask';

import { BankDetails } from '@deliveryapp/data-access';

import { StyledForm } from '../StyledForm';

interface BankDetailsFormProps {
  bankDetails?: BankDetails;
  handleChange: FormikHandlers['handleChange'];
}

export const BankDetailsForm: React.FC<BankDetailsFormProps> = ({
  bankDetails,
  handleChange
}) => {
  return (
    <StyledForm>
      <h3>Bank Details</h3>
      <div className="p-grid">
        <div className="p-col-12 row">
          <div className="input-wrapper p-float-label">
            <InputText
              id="name"
              type="text"
              name="bankDetails.name"
              data-testid="name"
              value={bankDetails?.name}
              onChange={handleChange}
            />
            <label htmlFor="name">Name</label>
            <i className="fa fa-building-o"></i>
          </div>
        </div>
        <div className="p-col-12 row">
          <div className="input-wrapper p-float-label">
            <InputMask
              id="accountNumber"
              type="text"
              name="bankDetails.accountNumber"
              data-testid="accountNumber"
              mask="9999 9999 9999 9999"
              unmask
              value={bankDetails?.accountNumber}
              onChange={handleChange}
            />
            <label htmlFor="accountNumber">Account Number</label>
            <i className="fa fa-credit-card"></i>
          </div>
        </div>
        <div className="p-col-12 row">
          <div className="input-wrapper p-float-label">
            <InputMask
              id="bin"
              type="text"
              name="bankDetails.bin"
              data-testid="bin"
              mask="9999999"
              unmask
              value={bankDetails?.bin}
              onChange={handleChange}
            />
            <label htmlFor="bin">BIN</label>
            <i className="fa fa-credit-card"></i>
          </div>
        </div>
        <div className="p-col-12 row">
          <div className="input-wrapper p-float-label">
            <InputMask
              id="swift"
              type="text"
              name="bankDetails.swift"
              data-testid="swift"
              mask="********"
              unmask
              value={bankDetails?.swift}
              onChange={handleChange}
            />
            <label htmlFor="swift">SWIFT</label>
            <i className="fa fa-credit-card"></i>
          </div>
        </div>
      </div>
    </StyledForm>
  );
};
