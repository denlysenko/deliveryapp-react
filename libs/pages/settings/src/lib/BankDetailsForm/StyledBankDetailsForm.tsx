import styled from 'styled-components';

export const StyledBankDetailsForm = styled.div`
  color: pink;
`;
