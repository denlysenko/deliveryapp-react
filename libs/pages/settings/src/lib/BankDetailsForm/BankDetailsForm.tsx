import React, { useRef } from 'react';

import { Growl } from 'primereact/growl';

import { StyledBankDetailsForm } from './StyledBankDetailsForm';

export const BankDetailsForm = () => {
  const growl = useRef<Growl>(null);

  return (
    <StyledBankDetailsForm>
      <Growl ref={growl} />
      <h1>Welcome to BankDetailsForm component!</h1>
    </StyledBankDetailsForm>
  );
};
