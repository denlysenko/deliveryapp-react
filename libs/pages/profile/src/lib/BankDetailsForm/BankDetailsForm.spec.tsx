import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { BankDetailsForm } from './BankDetailsForm';

const bankDetails = {
  name: '',
  accountNumber: '',
  bin: '',
  swift: ''
};

const handleChange = jest.fn();

describe('[Profile] BankDetailsForm', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render bank details form', () => {
    const { baseElement } = render(
      <BankDetailsForm bankDetails={bankDetails} handleChange={handleChange} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
