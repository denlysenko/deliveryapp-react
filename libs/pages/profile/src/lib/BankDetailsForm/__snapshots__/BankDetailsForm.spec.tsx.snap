// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Profile] BankDetailsForm should render password form 1`] = `
<body>
  <div>
    <div
      class="sc-AykKC jlhElc"
    >
      <h3>
        Bank Details
      </h3>
      <div
        class="p-grid"
      >
        <div
          class="p-col-12 row"
        >
          <div
            class="input-wrapper p-float-label"
          >
            <input
              class="p-inputtext p-component"
              data-testid="name"
              id="name"
              name="bankDetails.name"
              type="text"
              value=""
            />
            <label
              for="name"
            >
              Name
            </label>
            <i
              class="fa fa-building-o"
            />
          </div>
        </div>
        <div
          class="p-col-12 row"
        >
          <div
            class="input-wrapper p-float-label"
          >
            <input
              class="p-inputtext p-component"
              id="accountNumber"
              name="bankDetails.accountNumber"
              type="text"
            />
            <label
              for="accountNumber"
            >
              Account Number
            </label>
            <i
              class="fa fa-credit-card"
            />
          </div>
        </div>
        <div
          class="p-col-12 row"
        >
          <div
            class="input-wrapper p-float-label"
          >
            <input
              class="p-inputtext p-component"
              id="bin"
              name="bankDetails.bin"
              type="text"
            />
            <label
              for="bin"
            >
              BIN
            </label>
            <i
              class="fa fa-credit-card"
            />
          </div>
        </div>
        <div
          class="p-col-12 row"
        >
          <div
            class="input-wrapper p-float-label"
          >
            <input
              class="p-inputtext p-component"
              id="swift"
              name="bankDetails.swift"
              type="text"
            />
            <label
              for="swift"
            >
              SWIFT
            </label>
            <i
              class="fa fa-credit-card"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
